### PR TITLE
Gives clown cars headlights which are rainbow colored if emagged

### PR DESCRIPTION
**Pull request by ArcaneDefence**
## About The Pull Request
Uses directional lighting as a toggle via an action that only clown car drivers get
Clown cars process now too

## Why It's Good For The Game
**Pros:** 

1. Clowns can see where they're driving
2. Clowns get another button to click and make funny noises to annoy everyone
3. It makes sense to give cars headlights
4. If the car is emagged it flashes random colors of the rainbow

**Cons:**

1. Clowns can see where they're driving
2. Clowns get another button to click and make that goddamn fucking click never stop
3. GODDAMN THIS RAINBOW COLORED ABOMINATION

Video of emagged headlights- https://streamable.com/7ez8gp

image
Big light for big energy

## Changelog
:cl:
add: Clown cars are now fitted with functioning headlights! Emag them for some more high speed low drag variants!
/:cl:

### CODE REVIEW

**Comment by maxymax13** [comment]
why is icondiffbot missing

**Comment by TheVekter** [comment]
Can we make it so emagging the car makes the headlights cycle through the spectrum?

**Comment by ArcaneDefence** [comment]
> 
> 
> Can we make it so emagging the car makes the headlights cycle through the spectrum?

image

File: code/modules/vehicles/cars/clowncar.dm [hunk 1]
```diff
@@ -9,16 +9,32 @@
...
 	var/thankscount
 	var/cannonmode = FALSE
 	var/cannonbusy = FALSE
 
+/obj/vehicle/sealed/car/clowncar/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj,src)
+
+/obj/vehicle/sealed/car/clowncar/process()
+	if(light_on && obj_flags & EMAGGED )
```
Comment (Mothblocks) [inline code/modules/vehicles/cars/clowncar.dm:29]:
```suggestion
	if(light_on && (obj_flags & EMAGGED))
```

Comment (Mothblocks) [reply code/modules/vehicles/cars/clowncar.dm:29]: My concerns more with the trailing space, but I just like parentheses around bitops.

File: code/modules/vehicles/cars/clowncar.dm [hunk 2]
```diff
@@ -9,16 +9,32 @@
...
+		set_light_color(pick(headlight_colors))
+	else
+		return
```
Comment (Mothblocks) [inline code/modules/vehicles/cars/clowncar.dm:32]:
```suggestion
```

Return is implicit.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Minor nitpick.

**Pull request merged**